test/dashboard/index.spec.js

```diff
@@ -25,4 +26,126 @@ describe("dashboard", () => {
     expect(dashboardWithOptions.minimal).to.be.true;
   });
+
+  describe("set* methods", () => {
+    let dashboard;
+
+    beforeEach(() => {
+      dashboard = new Dashboard();
+    });
+
+    describe("setProgress", () => {
+      it("can setProgress", () => {
+        const data = {
+          value: 0.57
+        };
+
+        expect(dashboard.setProgress(data)).to.not.throw;
+        expect(dashboard.progressbar.setProgress).to.have.been.calledOnce;
+        expect(dashboard.progressbar.setContent).to.have.been.called;
+      });
+
+      it("should call progressbar.setProgress twice if not in minimal mode and percent is falsy", () => {
+        const data = {
+          value: null
+        };
+
+        expect(dashboard.setProgress(data)).to.not.throw;
+        expect(dashboard.progressbar.setProgress).to.have.been.calledTwice;
+      });
+    });
+
+    describe("setStats", () => {
+      const data = {
+        value: {
+          errors: null,
+          data: {
+            errors: [],
+            warnings: []
+          }
+        }
+      };
+
+      it("can setStats", () => {
+        expect(dashboard.setStats(data)).not.to.throw;
+        expect(dashboard.logText.log).to.have.been.called;
+        expect(dashboard.modulesMenu.setLabel)
+          .to.have.been.calledWith(chalk.yellow("Modules (loading...)"));
+        expect(dashboard.assets.setLabel)
+          .to.have.been.calledWith(chalk.yellow("Assets (loading...)"));
+        expect(dashboard.problemsMenu.setLabel)
+          .to.have.been.calledWith(chalk.yellow("Problems (loading...)"));
+      });
+
+      it("should display stats errors if present", () => {
+        data.value.errors = ["error"];
+
+        expect(dashboard.setStats(data)).not.to.throw;
+        expect(dashboard.status.setContent)
+          .to.have.been.calledWith("{red-fg}{bold}Failed{/}");
+      });
+    });
+
+    it("can setSizes", () => {
+      const data = {
+        value: {
+          assets: {
+            foo: {
+              meta: {
+                full: 456
+              }
+            },
+            bar: {
+              meta: {
+                full: 123
+              }
+            }
+          }
+        }
+      };
+
+      const formattedData = [
+        ["Name", "Size"],
+        ["foo", "456 B"],
+        ["bar", "123 B"],
+        ["Total", "579 B"]
+      ];
+
+      expect(dashboard.setSizes(data)).to.not.throw;
+      expect(dashboard.assets.setLabel).to.have.been.calledWith("Assets");
+      expect(dashboard.assetTable.setData).to.have.been.calledWith(formattedData);
+      expect(dashboard.modulesMenu.setLabel).to.have.been.calledWith("Modules");
+      expect(dashboard.modulesMenu.setItems).to.have.been.called;
+      expect(dashboard.modulesMenu.selectTab)
+        .to.have.been.calledWith(dashboard.modulesMenu.selected);
+      expect(dashboard.screen.render).to.have.been.called;
+    });
+
+    it("can setProblems", () => {
```

Comment:
I'm looking at this and will try to get to a bit more by sometime on Tuesday...

Reply:
I think something like this may get you where you want to be:

```diff
diff --git a/test/dashboard/index.spec.js b/test/dashboard/index.spec.js
index 6c9dd5b..4cf3a98 100644
--- a/test/dashboard/index.spec.js
+++ b/test/dashboard/index.spec.js
@@ -1,8 +1,9 @@
 "use strict";
 
 const chalk = require("chalk");
-require("../base.spec");
+const base = require("../base.spec");
 
+const blessed = require("blessed");
 const Dashboard = require("../../dashboard");
 
 describe("dashboard", () => {
@@ -122,6 +123,21 @@ describe("dashboard", () => {
     });
 
     it("can setProblems", () => {
+      // Override ListBar fakes from what we do in `base.spec.js`.
+      blessed.listbar.returns({
+        selected: "selected",
+        setLabel: base.sandbox.spy(),
+        setProblems: base.sandbox.spy(),
+        selectTab: base.sandbox.spy(),
+        setItems: base.sandbox.stub().callsFake((obj) => {
+          // Naively simulate what setItems would do calling each object key.
+          Object.keys(obj).forEach((key) => obj[key]());
+        })
+      });
+
+      // Discard generic dashboard, create a new one with adjusted mocks.
+      dashboard = new Dashboard();
+
       const data = {
         value: {
           duplicates: {
```

We're really pushing the boundaries of how much to mock, but here I think it may make sense because we really want to ensure those imports are (1) happening and (2) used.

Let me know if you need tips on getting even further in to things like asserting on what's called within `formatProblems` or whatnot...